### PR TITLE
SFH-19: Query Users

### PR DESCRIPTION
**Pull request by cristianoprogramador**
I created the test for the query users, using jest and a mock list of two users for you to comment and approve.

### CODE REVIEW

File: jest.setup.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { UserRepositoryMock } from "./src/domain/Users/repositories/repositoryMock";
+import {
+  GetProjectsByUserIdService,
+  ListUsersService,
+} from "./src/domain/Users/services";
+
+container
+  .registerSingleton<IUserRepository>("IUserRepository", UserRepositoryMock)
+  .register<ListUsersService>("ListUsersService", ListUsersService)
+  .registerSingleton<IUserRepository>("IUserRepository", UserRepositoryMock)
```
Comment (RenatoTMLima) [inline jest.setup.ts]: Why did you register the same stuff twice? Could you please, remove this one?

File: src/domain/Users/services/getProjectsByUserId.service.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import "reflect-metadata";
```
Comment (RenatoTMLima) [inline src/domain/Users/services/getProjectsByUserId.service.test.ts]: Why are you importing it here, if you already setup this import in the jest.setup file? Could you please remove it?

File: src/domain/Users/services/getProjectsByUserId.service.test.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { container } from "tsyringe";
+import { PROJECTS_LIST_MOCK } from "../repositories/repositoryMock";
+import { type GetProjectsByUserIdService } from "./getProjectsByUserId.service";
+
+describe("Get Projects By User ID", () => {
```
Comment (RenatoTMLima) [inline src/domain/Users/services/getProjectsByUserId.service.test.ts]: Could you please change the describe title to something with a patter? For example
```suggestion
describe("service: GetProjectsByUserIdService", () => {
```

File: src/domain/Users/services/listUsers.service.test.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { container } from "tsyringe";
+import { USERS_LIST_MOCK } from "../repositories/repositoryMock";
+import { type ListUsersService } from "./listUsers.service";
+
+describe("List of Users", () => {
```
Comment (RenatoTMLima) [inline src/domain/Users/services/listUsers.service.test.ts]: Could you replace this as well to follow a pattern?
```suggestion
describe("service: ListUsersService", () => {
```

File: src/domain/Users/services/getProjectsByUserId.service.test.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  it("should return projects associated with a user", async () => {
+    const service = container.resolve<GetProjectsByUserIdService>(
+      "GetProjectsByUserIdService"
+    );
+    const userId = 1;
+    const result = await service.execute(userId);
+
+    expect(result).toEqual(PROJECTS_LIST_MOCK);
```
Comment (RenatoTMLima) [inline src/domain/Users/services/getProjectsByUserId.service.test.ts]: Could you please use .toStrictEqual? This function makes a better evaluation when comparing structures like objects and arrays

File: src/domain/Users/services/getProjectsByUserId.service.test.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  it("should return an empty array if the user has no projects", async () => {
+    const service = container.resolve<GetProjectsByUserIdService>(
+      "GetProjectsByUserIdService"
+    );
+
+    const userId = 3;
+    const result = await service.execute(userId);
+
+    expect(result).toEqual([]);
```
Comment (RenatoTMLima) [inline src/domain/Users/services/getProjectsByUserId.service.test.ts]: Could you replace for toStrictEqual as well in here?

**Review by RenatoTMLima — CHANGES_REQUESTED** [review]

File: jest.setup.ts (hunk 1 above)
Comment (cristianoprogramador) [reply jest.setup.ts]: Wow, sorry about that. I wrote that while I was talking to you.

File: src/domain/Users/services/getProjectsByUserId.service.test.ts (hunk 2 above)
Comment (cristianoprogramador) [reply src/domain/Users/services/getProjectsByUserId.service.test.ts]: Yes, thank you for that.

File: src/domain/Users/services/getProjectsByUserId.service.test.ts (hunk 3 above)
Comment (cristianoprogramador) [reply src/domain/Users/services/getProjectsByUserId.service.test.ts]: Yes i can, thank you

File: src/domain/Users/services/getProjectsByUserId.service.test.ts (hunk 5 above)
Comment (cristianoprogramador) [reply src/domain/Users/services/getProjectsByUserId.service.test.ts]: yes i can

File: src/domain/Users/services/getProjectsByUserId.service.test.ts (hunk 6 above)
Comment (cristianoprogramador) [reply src/domain/Users/services/getProjectsByUserId.service.test.ts]: yes i can

**Review by RenatoTMLima — APPROVED** [review]

**Pull request merged**